modules/core-mvel/src/main/java/org/mule/runtime/core/internal/el/mvel/MVELExpressionExecutor.java

```diff
@@ -48,7 +54,16 @@ public MVELExpressionExecutor(final ParserConfiguration parserConfiguration) {
     this.parserConfiguration = parserConfiguration;
 
     MVEL.COMPILER_OPT_PROPERTY_ACCESS_DOESNT_FAIL = true;
-    OptimizerFactory.setDefaultOptimizer(OptimizerFactory.SAFE_REFLECTIVE);
+    // Since the DynamicOptimizer class grabs the TCCL, it could grab the app's classloader and that could lead to memory leakages
+    Thread currentThread = currentThread();
+    ClassLoader originalClassLoader = currentThread.getContextClassLoader();
+    ClassLoader newClassLoader = MVELExpressionLanguage.class.getClassLoader();
+    setContextClassLoader(currentThread, originalClassLoader, newClassLoader);
+    try {
+      OptimizerFactory.setDefaultOptimizer(OptimizerFactory.SAFE_REFLECTIVE);
+    } finally {
+      setContextClassLoader(currentThread, newClassLoader, originalClassLoader);
+    }
```

Comment:
I would move this to another PR. The reason being the following: this is a change in productive code, everything else affects only the tests. If there happens to be a bug in production with this change, it may require a revert followed by an alternative fix. By tying everything together into a single commit you are making it harder to revert cleanly.
Additionally, I would suggest adding a test that breaks without this fix that would show how the CL is potentially leaked. I know we already have a test that fails in such case but the problem is that test only fails if it is executed first in the test case, which is currently not guaranteed.
Finally, consider using `ClassUtils.withContextClassLoader` for readability and to reduce duplication.

Reply:
Got it, I'll move this change to another PR. I didn't use `ClassUtils.withContextClassLoader` since it's deprecated and using `setContextClassloader` instead is [recommended](https://github.com/mulesoft/mule/blob/a604d3e17ffab48397bb19ffbb4de27d714807d0/core/src/main/java/org/mule/runtime/core/api/util/ClassUtils.java#L967).